### DISCUSSION TITLE
why do different `splitrule` settings for competing risk give the same results?

### DISCUSSION THREAD

**Discussion by ringo-alin** [kogalur/randomForestSRC#163] [Q&A]
I'm trying to understand the "competing risk" vignette.  

(1) I wonder what other columns of the model's outcomes mean when  the model is set to be event-specific;

> Note importantly that when extracting VIMP from Analyses 2 and 3, we were careful to only extract VIMP for the targeted specific event. Thus for Analysis 2, this is column one of VIMP, which corresponds to death, whereas for Analysis 3, it is column two, which corresponds to transplant.

(2)  As the following example code shows,  I tried different `splitrule` settings and the model reacted only on `cause`. Are the results supposed to be the same?

My rsf version is 2.12.0. (BTW, version 3.0.0 seems to have a VIMP reproducibility problem #67)

```
library(randomForestSRC)
library(survival)

data(pbc, package = "survival")

# cause = c(1, 1)
o1 <- rfsrc(Surv(time, status) ~ ., pbc, seed = 1)
o2 <- rfsrc(Surv(time, status) ~ ., pbc, seed = 1, splitrule = "logrank")
# cause = c(1, 0)
o3 <- rfsrc(Surv(time, status) ~ ., pbc, seed = 1, splitrule = "logrankCR", cause = c(1, 0))
o4 <- rfsrc(Surv(time, status) ~ ., pbc, seed = 1, splitrule = "logrank", cause = c(1, 0))

# same error rate, outcome, prediction, vimp
o1$ensemble == o2$ensemble
o3$ensemble == o4$ensemble

all(predict(o1, pbc[1, ])$cif == predict(o2, pbc[1, ])$cif)
all(predict(o3, pbc[1, ])$cif == predict(o4, pbc[1, ])$cif)

all(vimp(o1)$importance == vimp(o2)$importance)
all(vimp(o3)$importance == vimp(o4)$importance)
```

**Comment by ishwaran** [comment]
Hi, thanks for your post.  It looks like this is a bug and we are looking into it.  We'll keep you updated.
